cupy/random/_generator_api.pyx

```diff
@@ -152,6 +152,10 @@ class Generator:
         if out is not None:
             self._check_output_array(dtype, size, out)
 
+        dtype_ = numpy.dtype(dtype)
```

Comment:
I think if desired `dtype` is neither float64/float32, it's better to generate using float32 and down cast to the dtype.